pkg/reconciler/isbsvc/installer/watcher.go

```diff
@@ -8,29 +8,29 @@ import (
 
 // getStatefulSetStatus returns a message describing statefulset status, and a bool value indicating if the status is considered done.
 // Borrowed at kubernetes/kubectl/rollout_status.go https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/rollout_status.go#L130
-func getStatefulSetStatus(sts *appv1.StatefulSet) (string, bool) {
+func getStatefulSetStatus(sts *appv1.StatefulSet) (string, string, bool) {
 	if sts.Status.ObservedGeneration == 0 || sts.Generation > sts.Status.ObservedGeneration {
-		return "Waiting for statefulset spec update to be observed...\n", false
+		return "Waiting for statefulset spec update to be observed...\n", "Progressing", false
+	}
+	if sts.Status.UpdateRevision != sts.Status.CurrentRevision {
+		return fmt.Sprintf("waiting for statefulset rolling update to complete %d pods at revision %s...\n",
+			sts.Status.UpdatedReplicas, sts.Status.UpdateRevision), "Progressing", false
 	}
 	if sts.Spec.Replicas != nil && sts.Status.ReadyReplicas < *sts.Spec.Replicas {
-		return fmt.Sprintf("Waiting for %d pods to be ready...\n", *sts.Spec.Replicas-sts.Status.ReadyReplicas), false
+		return fmt.Sprintf("Waiting for %d pods to be ready...\n", *sts.Spec.Replicas-sts.Status.ReadyReplicas), "Unavailable", false
 	}
 	if sts.Spec.UpdateStrategy.Type == appv1.RollingUpdateStatefulSetStrategyType && sts.Spec.UpdateStrategy.RollingUpdate != nil {
 		if sts.Spec.Replicas != nil && sts.Spec.UpdateStrategy.RollingUpdate.Partition != nil {
 			if sts.Status.UpdatedReplicas < (*sts.Spec.Replicas - *sts.Spec.UpdateStrategy.RollingUpdate.Partition) {
 				return fmt.Sprintf(
 					"Waiting for partitioned roll out to finish: %d out of %d new pods have been updated...\n",
-					sts.Status.UpdatedReplicas, *sts.Spec.Replicas-*sts.Spec.UpdateStrategy.RollingUpdate.Partition), false
+					sts.Status.UpdatedReplicas, *sts.Spec.Replicas-*sts.Spec.UpdateStrategy.RollingUpdate.Partition), "Unavailable", false
```

Comment:
It sounds like you changed this from "Progressing" to "Unavailable" based on your testing? Can you clarify the testing that caused you to change that? Did you find that you hit this when not in the initial startup phase? (If so, I guess the message "Waiting for partitioned roll out to finish: %d out of %d new pods have been updated" should probably be changed.)

Reply:
Ohh, It was my bad, I didn't mean to change it here, Let me update it